calblueprint/berkeley-student-cooperative
# Bug/user context

# ✨ New in this PR ✨

## One liner: what did you do? 🛠
Added loading to the returned elements so that authUserContext.Provider can be signaled when fetching the user has been completed; also added `await` to line 76 in auth.tsx so that the `userFirebaseAuth()` would wait for `establishUserContext()` to finish updated the user.


## Coverage 🙆‍♀️
_use this section to break up your task into submodules and track progress_ 
 - [x ] User persists through refreshing the page
 - [x ] Components can call the user context and use it

## How can the reviewer test your code? 👩‍💻
Open up the console to check for errors and also print statements that show the user's info. Log in, refresh the page, and check that the user's components show the correct information. Can be done using testing_create_account@gmail.com member account. Check the member's settings page and it should show Grg's member profile.

## Any bugs you encountered or still having trouble with? 🐛
When I am logged in (using testing_create_account@gmail.com's member account), I get the error below when I try to go back to `localhost:3000/`.
<img width="591" alt="image" src="https://user-images.githubusercontent.com/66578584/204663158-ae231506-25d7-441b-b002-34ce27790e90.png">

## Resources 📔
_link online resources and related PRs_ 
Danashi's PRs for User Context.

🏚 cc: @gregoriiaaa
